src/hist.jl

```diff
@@ -1,3 +1,69 @@
+# ---------------------------------------------------------------------------
+# Series recipe for a 1D plot of binned data via path. Could move into
+# Plot.jl at some point, so that backends can support it explicitly (if
+# doing so offers advantages over using the recipe).
+
+_hist_edge_centers(edge::AbstractVector) = (edge[2:end] + edge[1:end-1]) / 2
+
+_hist_outline_path(edge, weights) = begin
+    nbins = length(weights)
+    assert(length(edge) == nbins + 1)
+
+    it_e, it_w = start(edge), start(weights)
+    px, it_e = next(edge, it_e)
+    py = zero(eltype(weights))
+
+    npathpts = 2 * nbins + 2
+    x = Vector{eltype(px)}(npathpts)
+    y = Vector{eltype(py)}(npathpts)
+
+    x[1], y[1] = px, py
+    i = 2
+    while (i < npathpts - 1)
+        py, it_w = next(weights, it_w)
+        x[i], y[i] = px, py
+        i += 1
+        px, it_e = next(edge, it_e)
+        x[i], y[i] = px, py
+        i += 1
+    end
+    assert(i == npathpts)
+    x[end], y[end] = px, zero(py)
+
+    (x, y)
+end
+
+@recipe function f(::Type{Val{:binned}}, x, y, z)
+    edge, weights = x, y
+
+    axis = d[:subplot][Plots.isvertical(d) ? :xaxis : :yaxis]
+
+    xpts, ypts = _hist_outline_path(edge, weights)
+    if !Plots.isvertical(d)
+        xpts, ypts = ypts, xpts
+    end
+
+    # create a secondary series for the markers
+    if d[:markershape] != :none
+        @series begin
+            seriestype := :scatter
+            x := _hist_edge_centers(edge)
+            y := weights
+            fillrange := nothing
+            label := ""
+            primary := false
+            ()
+        end
+        markershape := :none
+    end
+
+    x := xpts
+    y := ypts
+    seriestype := :path
```

Comment:
cool enough. I just noticed Tom had used Segments and :shape for the boxplot recipe